main.go

```diff
@@ -48,18 +50,24 @@ const UserAgent = "cloud-run-proxy/" + Version + " (" + OSArch + ")"
 var (
 	flagHost             = flag.String("host", "", "Cloud Run host for which to proxy")
 	flagBind             = flag.String("bind", "127.0.0.1:8080", "local host:port on which to listen")
+	flagAudience         = flag.String("audience", "", "override JWT audience value (aud)")
 	flagToken            = flag.String("token", "", "override OIDC token")
 	flagPrependUserAgent = flag.Bool("prepend-user-agent", true, "prepend a custom User-Agent header to requests")
 )
 
 func main() {
-	if err := realMain(); err != nil {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	if err := realMain(ctx); err != nil {
+		cancel()
```

Comment:
Will `cancel()` be called twice since there is `defer cancel()` before?

Reply:
Yea, but that's okay. We need the `defer` to cancel the context in case of panics. We can't rely on the defer all the time though, because `os.Exit` exits _immediately_ and defers are not run. Cancelling twice is not an issue: https://go.dev/play/p/aqDO8v1GvJJ